sdk/core/azure-core-amqp/src/models/message_source.cpp

```diff
@@ -19,7 +19,10 @@
 #include <azure_uamqp_c/amqpvalue.h>
 #elif ENABLE_RUST_AMQP
 #include "rust_amqp_wrapper.h"
+
+#include <azure/core/amqp/internal/common/runtime_context.hpp>
 using namespace Azure::Core::Amqp::_detail::RustInterop;
```

Comment:
Previously, we'd have `_detail` and `_internal` as terminal namespace, we may even have a guideline for that. I see why it might make sense to have namespaces inside `_detail` and `_internal`, I'd just check that ApiView and doxygen don't have problems with them, if you'd prefer to keep it as `Azure::Core::Amqp::_detail::RustInterop` and not `Azure::Core::Amqp::RustInterop::_detail`.

Reply:
We have had this discussion before - but in this case, I get to specify the namespace for the RustInterop rust module. I don't get to put RustInterop in the middle of a namespace declaration.

So this naming is unfortunately the best we can do for Rust interoperability.

Reply:
For instance, if we follow the _detail as a terminal namespace, it leads to the impression that there might be types in the RustInterop namespace which are public. But the entire RustInterop namespace is an implementation detail of the AMQP stack. It can *never* be a part of the public API surface.

Reply:
I would not disagree with you, plus we could say, until there is no `Azure::Widget::X` namespace, there can be `Azure::Widget::_detail::X` namespace. After that, it becomes confusing, whether it is `X::_detail`, or `_detail::X`, plus `Azure::Widget::_detail::X` would have to refer to Azure::Widget::X::Class` as `X::Class`, while `Azure::Widget::X::_detail` does no need any prefix.
My comment was more to bring your attention. I don't feel strong this or the other way. I like that the `RustInterop` is not exposed. Just saying, it is probably against the current guidelines, but we could as well update the guidelines. I remember the time we discussed them, there was no strong reason really, we just decided, let's do `X::_detail` until we find out we need `_detail::X`. Maybe your case is exactly such that we rather need `_detail::X`.